### PR TITLE
Merge SQL modules

### PR DESCRIPTION
**Pull request by gierlachg**
Merged `hazelcast-sql`, `hazelcast-sql-core` and `hazelcast-jet-sql` modules into one - `hazelcast-sql`.

### CODE REVIEW

**Review by alex-dukhno — APPROVED** [review]

**Review by viliam-durina — APPROVED** [review]
As far as I can tell it's OK, though I didn't check if everything was transferred from jet-sql's pom to hazelcast-sql's pom.
But it would be good if the build master @frant-hartm has a look too before we merge.

**Review by frant-hartm — APPROVED** [review]
LGTM, couple of comments:

The slf4j inside calcite is relocated and it uses noop logger. Won't this be an issue for debugging?

When we allow users to extend SQL (e.g. with custom functions, sources / sinks) won't they face the same issues with relocated calcite classes as we had between `hazelcast-sql` and `hazelcast-jet-sql`? Or do we solve that when we actually provide such public API?

File: hazelcast-sql/pom.xml [hunk 1]
```diff
@@ -45,6 +60,101 @@
...
+                            <lookAhead>1</lookAhead>
+                            <isStatic>false</isStatic>
+                            <outputDirectory>${project.build.directory}/generated-sources/java/</outputDirectory>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-jar-plugin</artifactId>
```
[the remark this answers is not in the document]
Comment (gierlachg) [reply hazelcast-sql/pom.xml:160]: Enterprise uses it I believe.

**Comment by gierlachg** [comment]
> The slf4j inside calcite is relocated and it uses noop logger. Won't this be an issue for debugging?

I don't think we rely on `Calcite` logging anywhere - at least I didn't find it useful or miss it so far.

> When we allow users to extend SQL...

Hopefully, user facing APIs won't depend on Calcite. The other thing we are not really sure how is it going to look like so let's worry about it when time comes.

**Pull request merged**